code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/SystemStackUpdate.java

```diff
@@ -137,13 +145,44 @@ protected boolean addStack(Account account, String stack) throws IOException {
     }
 
     protected String getFile(String stack, String filename) throws IOException {
```

Comment:
I would change the response type of this method to just `Template`.  No need to make two API calls to `rancher-catalog-service` for the same thing.